lumeer-storage/lumeer-storage-mongodb/src/test/java/io/lumeer/storage/mongodb/dao/system/MongoOrganizationDaoTest.java

```diff
@@ -102,31 +120,135 @@ public void testCreateOrganization() {
             .isInstanceOf(ResourceNotFoundException.class)
             .hasFieldOrPropertyWithValue("resourceType", ResourceType.ORGANIZATION);
    }
 
+   @Test
+   public void testGetOrganizations() {
+      Organization organization = prepareOrganization(CODE1);
+      datastore.save(organization);
+
+      Organization organization2 = prepareOrganization(CODE2);
+      datastore.save(organization2);
+
+      DatabaseQuery query = new DatabaseQuery.Builder(USER).build();
+      List<Organization> organizations = organizationDao.getOrganizations(query);
+      assertThat(organizations).extracting(Organization::getCode).containsOnly(CODE1, CODE2);
+   }
+
+   @Test
+   public void testGetOrganizationsNoReadRole() {
+      Organization organization = prepareOrganization(CODE1);
+      Permission userPermission = new MongoPermission(USER2, Collections.singleton(Role.CLONE.toString()));
+      organization.getPermissions().updateUserPermissions(userPermission);
+      datastore.save(organization);
+
+      Organization organization2 = prepareOrganization(CODE2);
+      Permission groupPermission = new MongoPermission(GROUP2, Collections.singleton(Role.SHARE.toString()));
+      organization2.getPermissions().updateGroupPermissions(groupPermission);
+      datastore.save(organization2);
+
+      DatabaseQuery query = new DatabaseQuery.Builder(USER2).groups(Collections.singleton(GROUP2)).build();
+      List<Organization> organizations = organizationDao.getOrganizations(query);
+      assertThat(organizations).isEmpty();
+   }
+
+   @Test
+   public void testGetOrganizationsGroupRole() {
+      Organization organization = prepareOrganization(CODE1);
+      datastore.save(organization);
+
+      Organization organization2 = prepareOrganization(CODE2);
+      datastore.save(organization2);
+
+      DatabaseQuery query = new DatabaseQuery.Builder(USER2).groups(Collections.singleton(GROUP)).build();
+      List<Organization> organizations = organizationDao.getOrganizations(query);
+      assertThat(organizations).extracting(Organization::getCode).containsOnly(CODE1, CODE2);
+   }
+
+   @Test
+   public void testDeleteOrganization() {
+      Organization organization = prepareOrganization(CODE1);
+      datastore.save(organization);
+      assertThat(organization.getId()).isNotNull();
+
+      organizationDao.deleteOrganization(organization.getCode());
+
+      Organization storedOrganization = datastore.createQuery(MongoOrganization.class)
+                     .field(MongoOrganization.CODE).equal(organization.getCode()).get();
+      assertThat(storedOrganization).isNull();
+   }
+
+   @Test
+   public void testDeleteOrganizationNotExisting() {
+      assertThatThrownBy(() -> organizationDao.deleteOrganization(NOT_EXISTING_CODE))
+            .isInstanceOf(WriteFailedException.class);
+   }
+
+   @Test
+   public void testEditOrganizationCode() {
+      MongoOrganization organization = (MongoOrganization) prepareOrganization(CODE1);
```

Comment:
You can change the return type of `prepareOrganization` method to `MongoOrganization` so you do not need to cast it every time.